backend/src/musicplatform/remotes/SpotifyRemote.ts

```diff
@@ -0,0 +1,167 @@
+import { SimplifiedArtist, SpotifyApi, Track } from "@spotify/web-api-ts-sdk";
+import { JSONTrack, PlayingJSONTrack } from "commons/backend-types";
+import { adminSupabase } from "../../server";
+import MusicPlatform from "../MusicPlatform";
+import Remote from "./Remote";
+
+export default class SpotifyRemote extends Remote {
+  spotifyClient: SpotifyApi;
+  musicPlatform: MusicPlatform;
+
+  protected constructor(
+    spotifyClient: SpotifyApi,
+    musicPlatform: MusicPlatform
+  ) {
+    super();
+    this.spotifyClient = spotifyClient;
+    this.musicPlatform = musicPlatform;
+  }
+
+  static async createRemote(
+    roomId: string,
+    musicPlatform: MusicPlatform
+  ): Promise<SpotifyRemote | null> {
+    const { data, error } = await adminSupabase
+      .from("rooms")
+      .select("*, user_profile(*, bound_services(*))")
+      .eq("id", roomId)
+      .single();
+
+    if (error) return null;
+    if (data.user_profile?.bound_services === undefined) return null;
+    if (data.user_profile?.bound_services.length === 0) return null;
```

Comment:
Prettier will make it readable I hope but just do it like this :
```suggestion
    if (error || data.user_profile?.bound_services === undefined || data.user_profile?.bound_services.length === 0) return null;
```